### PR TITLE
feature(#32): add functionality that select an item on the browser

### PR DESCRIPTION
**Pull request by neptune-web**
close ttt246/Brain/#32

### CODE REVIEW

File: src/background/index.mjs [hunk 1]
```diff
@@ -5,12 +5,27 @@ import { openUrl } from '../utils/open-url'
 import { registerPortListener } from '../services/wrappers.mjs'
 import { refreshMenu } from './menus.mjs'
 import { registerCommands } from './commands.mjs'
-import { getManagementBrowser } from '../services/apis/rising-api.mjs'
+import { getItemToSelect, getManagementBrowser } from '../services/apis/rising-api.mjs'
 
 async function executeApi(session, port, config) {
   // ### the part for integrating with backend ###
   console.debug('config', config)
```
Comment (stackdev37) [inline src/background/index.mjs:12]: Please remove this debug code:

File: src/background/index.mjs [hunk 2]
```diff
@@ -5,12 +5,27 @@ import { openUrl } from '../utils/open-url'
...
-  await getManagementBrowser(port, session.question, session, '')
+
+  console.log('browser-type--->', session.type)
```
Comment (stackdev37) [inline src/background/index.mjs:14]: Please remove this debug code:

File: src/components/ConversationCard/index.jsx [hunk 3]
```diff
@@ -181,13 +182,38 @@ function ConversationCard(props) {
...
+          case 'message':
+            updateAnswer(answer.content, false, 'answer')
+            break
+          case 'browser':
+            window.open(answer.content, '_blank', 'noreferrer')
+            break
+          case 'select_item_detail_info':
+            // eslint-disable-next-line no-case-declarations
+            const links = scrapeATags()
+            console.log('scrap_links----->', links)
```
Comment (stackdev37) [inline src/components/ConversationCard/index.jsx:194]: Please remove this debug code:

File: src/services/apis/rising-api.mjs [hunk 4]
```diff
@@ -121,3 +121,52 @@ export async function getManagementBrowser(port, question, session, apiKey) {
...
+
+export async function getItemToSelect(port, question, links, session, apiKey) {
+  const uuid = uuidv4()
+
+  const { controller, messageListener } = setAbortController(port)
+
+  const config = await getUserConfig()
+  const apiUrl = config.selectItemApiUrl
+
+  console.log('select_item_question ---->', question)
```
Comment (stackdev37) [inline src/services/apis/rising-api.mjs:133]: Please remove this debug code:

File: src/services/apis/rising-api.mjs [hunk 5]
```diff
@@ -121,3 +121,52 @@ export async function getManagementBrowser(port, question, session, apiKey) {
...
+      Authorization: `Bearer ${apiKey}`,
+    },
+    body: JSON.stringify({
+      token: 'test_token', // it is token generated in firebase for push notification. it can be ignored
+      uuid: uuid, // it is uuid for mobile devices or web users as well. please put it on "extension" for now.
+      items: links,
+      prompt: question,
+    }),
+    onMessage(message) {
+      console.debug('sse message', message)
```
Comment (stackdev37) [inline src/services/apis/rising-api.mjs:151]: Please remove this debug code:

File: src/services/apis/rising-api.mjs [hunk 6]
```diff
@@ -121,3 +121,52 @@ export async function getManagementBrowser(port, question, session, apiKey) {
...
+      if (message === '[DONE]') {
+        pushRecord(session, question, answer)
+        console.debug('conversation history', { content: session.conversationRecords })
```
Comment (stackdev37) [inline src/services/apis/rising-api.mjs:154]: Please remove this debug code:

File: src/utils/fetch-sse.mjs [hunk 7]
```diff
@@ -55,10 +55,12 @@ export async function fetchForBrowserMng(resource, options) {
   let hasStarted = false
   for await (const chunk of streamAsyncIterable(resp.body)) {
     const str = new TextDecoder().decode(chunk)
+    console.log('response data -------->', str)
```
Comment (stackdev37) [inline src/utils/fetch-sse.mjs:58]: Please remove this debug code:

File: src/utils/fetch-sse.mjs [hunk 8]
```diff
@@ -55,10 +55,12 @@ export async function fetchForBrowserMng(resource, options) {
...
     if (!str.startsWith('{') && !str.startsWith('"{')) {
       parser.feed(str)
     } else {
-      const result = JSON.parse(str).result
+      const result = JSON.stringify(JSON.parse(str).result)
+      console.log('response_result---->', result)
```
Comment (stackdev37) [inline src/utils/fetch-sse.mjs:63]: Please remove this debug code:

**Review by stackdev37 — CHANGES_REQUESTED** [review]

**Comment by neptune-web** [comment]
I've removed all consoles.

**Review by stackdev37 — APPROVED** [review]
Good job!

**Pull request merged**